uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature

```diff
@@ -1228,6 +1228,161 @@ Feature: GGMQ-1
+      | aws.greengrass.clientdevices.mqtt.EMQX   | LATEST                                  |
+      | aws.greengrass.clientdevices.IPDetector  | LATEST                                  |
+      | aws.greengrass.Cli                       | LATEST                                  |
+      | <agent>                                  | classpath:/local-store/recipes/<recipe> |
+    And I create client device "publisher"
+    And I create client device "subscriber"
+
+    And I update my Greengrass deployment configuration, setting the component aws.greengrass.clientdevices.Auth configuration to:
+    """
+{
+    "MERGE":{
+        "deviceGroups":{
+            "formatVersion":"2021-03-05",
+            "definitions":{
+                "MyPermissiveDeviceGroup":{
+                    "selectionRule":"thingName: ${publisher} OR thingName: ${subscriber}",
+                    "policyName":"MyPermissivePolicy"
+                }
+            },
+            "policies":{
+                "MyPermissivePolicy":{
+                    "AllowAll":{
+                        "statementDescription":"Allow client devices to perform all actions.",
+                        "operations":[
+                            "*"
+                        ],
+                        "resources":[
+                            "*"
+                        ]
+                    }
+                }
+            }
+        }
+    }
+}
+    """
+
+    And I update my Greengrass deployment configuration, setting the component <agent> configuration to:
+    """
+{
+    "MERGE":{
+        "controlAddresses":"${mqttControlAddresses}",
+        "controlPort":"${mqttControlPort}"
+    }
+}
+    """
+    When I associate "publisher" with ggc
+    When I associate "subscriber" with ggc
+
+    And I deploy the Greengrass deployment configuration
+    Then the Greengrass deployment is COMPLETED on the device after 5 minutes
+    And the aws.greengrass.clientdevices.mqtt.EMQX log on the device contains the line "is running now!." within 1 minutes
+
+    And I discover core device broker as "default_broker" from "publisher" in OTF
+    And I connect device "publisher" on <agent> to "default_broker" using mqtt "<mqtt-v>"
+    And I connect device "subscriber" on <agent> to "default_broker" using mqtt "<mqtt-v>"
+
+    When I subscribe "subscriber" to "iot_data_0" with qos 1 and expect status "<subscribe-status-q1>"
+    When I publish from "publisher" to "iot_data_0" with qos 1 and message "Test message0" and expect status 0
+    And message "Test message0" received on "subscriber" from "iot_data_0" topic within 10 seconds
+
+    # Reset CDA configuration
+    And I update my local deployment configuration, setting the component aws.greengrass.clientdevices.Auth configuration to:
+    """
+{
+   "RESET":[""]
+}
+    """
+    Then the local Greengrass deployment is SUCCEEDED on the device after 120 seconds
+
+    Then I wait 65 seconds
```

Comment:
Is it possible to replace wait to checking logs for some message?

Reply:
In that case have no reasons to wait for "Finished deployment task". We can wait until components you use actually restarted and ready to accept requests.  For example because EQMX started in docker it is possible when deployment finished and reported status, scenario continue to next steps but EMQX still booting and does not accept client's connection yet.

Reply:
There is no message in logs, which could definitely say that the Auth is ready. So I reverted changes in 7b413a7